src/App.jsx

```diff
@@ -104,23 +110,30 @@ function App() {
     return adjacentMinesCount;
   };
 
-  const openAllMines = (gameBoard) => {
-    let updatedBoard = gameBoard.map((row) => {
-      return row.map((tile) => {
-        if (tile.hasMine ) {
-          if ( ! tile.isFlagged ){
-            tile.isOpened = true;
-          }
-        }
-        else if ( tile.isFlagged ){
-          tile.isIncorrectlyFlagged = true
-        }
-        return tile;
-      });
-    });
+  const markIncorrectlyPlacedFlags = (gameBoard) => {
+    for ( const flagLocation of flagLocations ){
+      let tile = gameBoard[flagLocation.x][flagLocation.y];
 
-    return updatedBoard;
-  };
+      if ( ! tile.hasMine ){
+        tile.isIncorrectlyFlagged = true;
+      }
+    }
+
+    return gameBoard;
+  }
+
+  const revealUnflaggedMines = (gameBoard) => {
+    
+    for( const mineLocation of mineLocations ){
+      let tile = gameBoard[mineLocation.x][mineLocation.y];
+
+      if ( ! tile.isFlagged){
+        tile.isOpened = true;
+      }
+    }
+
+    return gameBoard;
```

Comment:
Here I have similar comment that above, let's avoid mutating the parameter gameBoard since this side-effect could cause issues in the future.

This can be done in different ways, you can even just make the clone of gameBoard inside this function instead of do it out of this function.

I like how it looks with the reduce since reduce already makes this for you.